### ISSUE TITLE
`parametrized` doesn't work properly with `pytest.param`

### ISSUE THREAD

**Issue by mathrick** [coady/pytest-parametrized#17]
If any of the values passed in are wrapped in `pytest.param`, the resulting values are not properly unpacked with `@parametrized` and thus change what the test actually receives.

Repro example:
```python
import pytest
from parametrized import parametrized

VALUES1 = [
    "foo",
]

VALUES2 = [
    "foo",
    pytest.param("baz", marks=pytest.mark.xfail(reason="it's different")),
]

@pytest.mark.parametrize("arg1", VALUES1)
@pytest.mark.parametrize("arg2", VALUES2)
def test_plain(arg1, arg2):
    assert arg1 == arg2

@parametrized.product
def test_parametrized(arg1=VALUES1, arg2=VALUES2):
    assert arg1 == arg2

```
Result:
```
============================================================== FAILURES ===============================================================
____________________________________________________ test_parametrized[arg21-foo] _____________________________________________________

arg1 = 'foo'
arg2 = ParameterSet(values=('baz',), marks=(MarkDecorator(mark=Mark(name='xfail', args=(), kwargs={'reason': "it's different"})),), id=None)

    @parametrized.product
    def test_parametrized(arg1=VALUES1, arg2=VALUES2):
>       assert arg1 == arg2
E       assert 'foo' == ParameterSet(values=('baz',), marks=(MarkDecorator(mark=Mark(name='xfail', args=(), kwargs={'reason': "it's different"})),), id=None)

/tmp/test.py:20: AssertionError
```

**Comment by coady** [comment]
The issue is `param` requires wrapping all the argument values, so it's impossible for this to work with multiple "zipped" values. Because the point of this library is to allow the values to be unzipped.

This particular case could be supported, because it's single arguments being stacked. So I'm willing to change `parametrized.product` so it simulates stacking, as long as it's clear that `parametrized.zip` can't support this.

**Comment by mathrick** [comment]
Thanks, I think this would be a reasonable compromise, especially together with a prominent note in the docs.

**Comment by mathrick** [comment]
Also, do you think it would make sense to "promote" the `pytest.param` to wrap the entire zipped value? If at most one of the values being zipped is wrapped, then I don't think it'd change the semantics in an unexpected way.

**Comment by coady** [comment]
Released [v1.6](https://github.com/coady/pytest-parametrized/releases/tag/v1.6).
